### PR TITLE
chore(deps): update dependency typescript to v5.5.3

### PR DESCRIPTION
**Pull request by Josh-Walker-GM**
See #10994.

### CODE REVIEW

File: packages/graphql-server/src/directives/makeDirectives.ts [hunk 1]
```diff
@@ -50,9 +50,9 @@ export const makeDirectivesForPlugin = (
 export const getDirectiveName = (schema: DocumentNode) => {
   const definition = schema.definitions.find(
     (definition) => definition.kind === 'DirectiveDefinition',
-  ) as ExecutableDefinitionNode
+  ) as ExecutableDefinitionNode | undefined
 
-  return definition.name?.value
+  return definition?.name?.value
 }
```
Comment (Josh-Walker-GM) [inline packages/graphql-server/src/directives/makeDirectives.ts:56]: You okay to sanity check that this is the appropriate tweak @Tobbe?

Comment (Tobbe) [reply packages/graphql-server/src/directives/makeDirectives.ts:56]: I'll have a look in a bit

File: packages/graphql-server/src/directives/makeDirectives.ts [hunk 2]
```diff
@@ -50,9 +50,9 @@ export const makeDirectivesForPlugin = (
 export const getDirectiveName = (schema: DocumentNode) => {
   const definition = schema.definitions.find(
     (definition) => definition.kind === 'DirectiveDefinition',
-  ) as ExecutableDefinitionNode
+  ) as ExecutableDefinitionNode | undefined
```
Comment (Tobbe) [inline packages/graphql-server/src/directives/makeDirectives.ts:53]: Looks like this cast isn't needed at all anymore
```suggestion
  )
```

File: packages/graphql-server/src/directives/makeDirectives.ts (hunk 1 above)
Comment (Tobbe) [reply packages/graphql-server/src/directives/makeDirectives.ts:56]: Yes, this looks good

File: packages/graphql-server/src/directives/makeDirectives.ts (hunk 2 above)
Comment (Josh-Walker-GM) [reply packages/graphql-server/src/directives/makeDirectives.ts]: I'll get rid of it then. Thanks!

**Review by Tobbe — APPROVED** [review]

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**